packages/forms/src/index.tsx

```diff
@@ -172,35 +226,82 @@ const JSONValidation = (value: Record<string, unknown> | null) => value !== null
  *
  * Otherwise we check to see if any of Redwood's `valueAs` props are present.
  */
+
+type SetCoersionProps = {
+  type?: string
+  name: string
+  emptyAsNull: boolean
+  emptyAsUndefined: boolean
+}
+
 const setCoercion = (
   validation: RedwoodRegisterOptions,
-  { type, name }: { type?: string; name: string }
+  { type, name, emptyAsNull, emptyAsUndefined }: SetCoersionProps
 ) => {
-  if (
-    validation.valueAsNumber ||
-    validation.valueAsDate ||
-    validation.setValueAs
-  ) {
+  if (validation.setValueAs) {
+    // Note, this case could overide other props
     return
   }
 
-  const valueAsProp = Object.keys(valueAsProps).find(
-    (valueAsProp) => valueAsProp in validation
-  )
-
-  if (valueAsProp) {
-    validation.setValueAs =
-      valueAsProps[valueAsProp as keyof typeof valueAsProps]
-    delete validation[valueAsProp as keyof typeof valueAsProps]
-    if (valueAsProp === 'valueAsJSON' && !validation.validate) {
-      validation.validate = JSONValidation
+  if (validation.valueAsBoolean || type === 'checkbox') {
+    // boolean case
+    if (emptyAsNull) {
+      validation.setValueAs = SET_VALUE_AS_FCNS.valueAsBoolean.emptyAsNull
+    } else if (emptyAsUndefined) {
+      validation.setValueAs = SET_VALUE_AS_FCNS.valueAsBoolean.emptyAsUndefined
+    } else {
+      validation.setValueAs = SET_VALUE_AS_FCNS.valueAsBoolean.base
+    }
+    delete validation.valueAsBoolean
+  } else if (validation.valueAsJSON) {
+    // JSON case
+    if (emptyAsNull) {
+      validation.setValueAs = SET_VALUE_AS_FCNS.valueAsJSON.emptyAsNull
+    } else if (emptyAsUndefined) {
+      validation.setValueAs = SET_VALUE_AS_FCNS.valueAsJSON.emptyAsUndefined
+    } else {
+      validation.setValueAs = validation.required
+        ? SET_VALUE_AS_FCNS.valueAsJSON.emptyAsDefaultRequired
+        : SET_VALUE_AS_FCNS.valueAsJSON.emptyAsNull
+    }
+    validation.validate = JSONValidation
+    delete validation.valueAsJSON
+  } else if (
+    type === 'date' ||
+    type === 'datetime-local' ||
+    validation.valueAsDate
+  ) {
+    // Date case
+    // Note we cannot pass on the standard r-h-f valueAsDate prop because
+    // the setValue technique is not compatible with r-h-f valueAsDate prop
+    // this prevents the use the emptyAsUndefined and emptyAsNull logic
+    // Thus, we need to implement our own value to date conversion
```

Comment:
```suggestion
    // Note we cannot pass on the standard r-h-f valueAsDate prop because
    // the setValue technique is not compatible with r-h-f valueAsDate prop.
    // This prevents the use of the emptyAsUndefined and emptyAsNull logic.
    // Thus, we need to implement our own value to date conversion
```